tests/transforms/test_transpile.py

```diff
@@ -114,6 +114,17 @@ def circuit():
         with pytest.raises(NotImplementedError, match=err_msg):
             transpiled_qnode()
 
+    def test_transpile_non_commuting_observables(self):
+        """Test that transpile will work with non-commuting observables."""
+
+        ops = (qml.CRX(0.1, wires=(0, 2)),)
+        ms = (qml.expval(qml.X(0)), qml.expval(qml.Y(0)))
+        tape = qml.tape.QuantumScript(ops, ms, shots=50)
+        [out], _ = qml.transforms.transpile(tape, coupling_map=((0, 1), (1, 2)))
+
+        expected = qml.tape.QuantumScript((qml.SWAP((1, 2)), qml.CRX(0.1, (0, 1))), ms, shots=50)
```

Comment:
Interesting, would have expected a "swap back" step in here :thinking: Is it skipped because the measurement is on `0`, which was never swapped? :thinking:

Reply:
Both measurements are on zero, which doesn't need to be changed. Otherwise, we would map their wires as well.